AdvanceIfNeeded on UnsetIterator can cause iterator to fail to advance

I ran this test:

```
func TestUnsetIteratorAdvanceWithCurrentValue(t *testing.T) {
	b := &Bitmap{}
	b.AddRange(0, 0x10000)
	iter := b.UnsetIterator(0, 0x10002)
	var got []uint32
	prev := uint32(0)
	for iter.HasNext() && len(got) < 10 {
		iter.AdvanceIfNeeded(prev)
		x := iter.Next()
		got = append(got, x)
		prev = x
	}
	assert.Equal(t, []uint{0x10000, 0x10001, 0x10002}, got)
}
```

And it failed as follows:

```
--- FAIL: TestUnsetIteratorAdvanceWithCurrentValue (0.00s)
    roaring_test.go:2851: 
        	Error Trace:	/home/rogpeppe/gohack/github.com/RoaringBitmap/roaring/roaring_test.go:2851
        	Error:      	Not equal: 
        	            	expected: []uint([]uint{0x10000, 0x10001, 0x10002})
        	            	actual  : []uint32([]uint32{0x10000, 0x10000, 0x10000, 0x10000, 0x10000, 0x10000, 0x10000, 0x10000, 0x10000, 0x10000})
        	Test:       	TestUnsetIteratorAdvanceWithCurrentValue
FAIL
```

It seems that `AdvanceIfNeeded` is not working as it should with `UnsetIterator` (similar logic with `Iterator` works fine).

This is on commit 2e9b4b9583f4603f6d5aad24a3fbd47b5ce5d08c

--- a/iter_test.go
+++ b/iter_test.go
@@ -417,6 +417,24 @@ func TestUnsetIteratorPeekable(t *testing.T) {
 		assert.False(t, it.HasNext())
 	})
 
+	t.Run("advance if needed on current value", func(t *testing.T) {
+		b := New()
+		b.AddRange(0, 0x10000)
+		iter := b.UnsetIterator(0, 0x10002)
+		var got []uint32
+		prev := uint32(0)
+		for len(got) < 10 {
+			iter.AdvanceIfNeeded(prev)
+			if !iter.HasNext() {
+				break
+			}
+			x := iter.Next()
+			got = append(got, x)
+			prev = x
+		}
+		assert.Equal(t, []uint32{0x10000, 0x10001, 0x10002}, got)
+	})
+
 	t.Run("peek next on empty iterator", func(t *testing.T) {
 		b := New()
 		b.AddInt(5) // Set bit in middle of range

--- a/roaring.go
+++ b/roaring.go
@@ -814,8 +814,8 @@ func (ui *unsetIterator) PeekNext() uint32 {
 
 // AdvanceIfNeeded advances the iterator so that the next value is at least minval
 func (ui *unsetIterator) AdvanceIfNeeded(minval uint32) {
-	if minval <= ui.min {
-		return // Already at or before the start of our range
+	if uint64(minval) <= ui.current {
+		return // Already at or past minval
 	}
 
 	if minval > ui.max {

